### PR TITLE
Add a `copy()` method to `Buffer`

### LINKED ISSUE #8102: [FEA] Provide a method to `copy()` a `cudf.core.buffer.Buffer()`
Likely we first need `rmm.DeviceBuffer` to expose a `copy()` method, which in turn could use [copy_device_to_ptr](https://github.com/rapidsai/rmm/blob/cab60bfcae9751d69efa148ae52bf9130704e51d/python/rmm/_lib/device_buffer.pyx#L450).

### PR DESCRIPTION
**Pull request by shwina**
Closes #8102

### CODE REVIEW

**Comment by shwina** [comment]
rerun tests

**Review by isVoid — APPROVED** [review]
lgtm with a small question

**Review by galipremsagar — APPROVED** [review]

File: python/cudf/cudf/tests/test_buffer.py [hunk 1]
```diff
@@ -44,3 +45,14 @@ def test_buffer_from_cuda_iface_dtype(data, dtype):
...
             buf = Buffer(data=data, size=data.size)  # noqa: F841
+
+
+@pytest.mark.parametrize("size", [0, 1, 10, 100, 1000, 10_000])
+def test_buffer_copy(size):
+    data = cp.random.randint(low=0, high=100, size=size, dtype="u1")
+    buf = Buffer(data=data)
+    got = buf.copy()
+    assert got.size == buf.size
+    if size > 0:
```
[the remark this answers is not in the document]
Comment (shwina) [reply python/cudf/cudf/tests/test_buffer.py:56]: Then `ptr` is `0` (nullptr), so the inequality check fails.

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**